qemu/tests/qemu_guest_agent_update.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import re
+import time
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import data_dir
+
+from qemu.tests.qemu_guest_agent import QemuGuestAgentBasicCheckWin
+
+
+class QemuGuestAgentUpdateTest(QemuGuestAgentBasicCheckWin):
+
+    @error_context.context_aware
+    def gagent_check_pkg_update(self, test, params, env):
+        """
+        Update qemu-ga-win pkg.
+
+        steps:
+            1)boot up guest.
+            2)install the previous qemu-ga in guest.
+                a.for virtio-win method,change to download iso.
+                b.for url method,need to finde the previous version and get
+                the download cmd.
+            3)update to the latest one.
+            4)qemu-ga basic test.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        def _change_agent_media(cdrom_virtio):
+            """
+            Only for virtio-win method,change virtio-win iso.
+
+            :param cdrom_virtio: iso file
+            """
+            logging.info("Change cdrom to %s" % cdrom_virtio)
+            virtio_iso = utils_misc.get_path(data_dir.get_data_dir(),
+                                             cdrom_virtio)
+            vm.change_media("drive_virtio", virtio_iso)
+
+            logging.info("Wait until device is ready")
+            vol_virtio_key = "VolumeName like '%virtio-win%'"
+            timeout = 10
+            end_time = time.time() + timeout
+            while time.time() < end_time:
+                time.sleep(2)
+                virtio_win_letter = utils_misc.get_win_disk_vol(session,
+                                                                vol_virtio_key)
+                if virtio_win_letter:
+                    break
+            if not virtio_win_letter:
+                test.fail("Couldn't get virtio-win volume letter.")
+
+        def _get_pkg_download_cmd():
+            """
+            Only for url method to get the previous qemu-ga version and get the
+            download cmd for this pkg.
+
+            """
+            qga_html = params["qga_html"]
```

Comment:
This is confusing, to name it `tmp_output` would be more meaningful, and it actually does not matter where the path is, so define it here directly with `tmp_output = /tmp/qemu-ga.html` is sufficient.